component/src/main/java/io/siddhi/extension/io/cdc/source/listening/MongoChangeDataCapture.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.io.cdc.source.listening;
+
+import io.siddhi.core.stream.input.source.SourceEventListener;
+import io.siddhi.extension.io.cdc.util.CDCSourceConstants;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * This class is for capturing change data for MongoDB using debezium embedded engine.
+ **/
+public class MongoChangeDataCapture extends ChangeDataCapture {
+
+    public MongoChangeDataCapture(String operation, SourceEventListener sourceEventListener) {
+        super(operation, sourceEventListener);
+    }
+
+    Map<String, Object> createMap(ConnectRecord connectRecord, String operation) {
+
+        //Map to return
+        Map<String, Object> detailsMap = new HashMap<>();
+
+        Struct record = (Struct) connectRecord.value();
+
+        //get the change data object's operation.
+        String op;
+
+        try {
+            op = (String) record.get(CDCSourceConstants.CONNECT_RECORD_OPERATION);
+        } catch (NullPointerException | DataException ex) {
```

Comment:
you do not have to catch null pointer exception please write it in the way this is not thrown and once you catch a exception if applicable log it

Reply:
'record' is returned from debezium. From that only extract the 'op' value.